### PR TITLE
Backend/requests + reviews

### LINKED ISSUE #32: Implement Request Service

### PR DESCRIPTION
**Pull request by okidokidokkaebi**
Please review and merge #52 first.

Resolves #32

### CODE REVIEW

File: src/main/java/com/peerrequest/app/services/DirectRequestProcessService.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.peerrequest.app.services;
+
+import com.peerrequest.app.data.DirectRequestProcess;
+
+public interface DirectRequestProcessService
+        extends CrudService<DirectRequestProcess, Long, DirectRequestProcess.Dto> {
+}
```
Comment (Wil-Lee) [inline src/main/java/com/peerrequest/app/services/DirectRequestProcessService.java:7]: claim open slot is missing

File: src/main/java/com/peerrequest/app/data/DirectRequest.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                .build();
+    }
+
+    /**
+     * Creates a DTO for the current request.
+     *
+     * @return DTO
+     */
+    public Dto toDto() {
+        // TODO: implement
```
Comment (Wil-Lee) [inline src/main/java/com/peerrequest/app/data/DirectRequest.java:61]: delete todo

**Comment by bits0rcerer** [comment]
Note: rebased `backend/requests` onto `backend/entries`

File: src/main/java/com/peerrequest/app/api/DirectRequestsController.java [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+     *
+     * @return directRequestProcess of entry
+     */
+    @GetMapping(value = "/categories/{category_id}/entries/{entry_id}/process", produces = "application/json")
+    public DirectRequestProcess.Dto getDirectRequestProcess(@PathVariable("entry_id") final Long entryId) {
+        var option = this.directRequestProcessService.get(entryId);
+        if (option.isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "direct request process does not exist");
+        }
+
```
Comment (Wil-Lee) [inline src/main/java/com/peerrequest/app/api/DirectRequestsController.java:37]: Check is missing, if the calling user is the creator of the entry

File: src/main/java/com/peerrequest/app/api/DirectRequestsController.java [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+    private final int maxPageSize = 100;
+
+    /**
+     * Gets a directRequestProcess.
+     *
+     * @param entryId entry of the directRequestProcess
+     *
+     * @return directRequestProcess of entry
+     */
+    @GetMapping(value = "/categories/{category_id}/entries/{entry_id}/process", produces = "application/json")
```
Comment (Wil-Lee) [inline src/main/java/com/peerrequest/app/api/DirectRequestsController.java:31]: remove:  produce = "aplication/json"
to match the other controllers

File: src/main/java/com/peerrequest/app/api/DirectRequestsController.java [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+
+    /**
+     * Creates a directRequestProcess.
+     *
+     * @param entryId entry of the directRequestProcess
+     * @param dto containing the open slots
+     *
+     * @return the added directRequestProcess
+     */
+    @PostMapping(value = "/categories/{category_id}/entries/{entry_id}/process", produces = "application/json")
```
Comment (Wil-Lee) [inline src/main/java/com/peerrequest/app/api/DirectRequestsController.java:49]: remove:  produce = "aplication/json"
to match the other controllers

File: src/main/java/com/peerrequest/app/api/DirectRequestsController.java [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+
+    /**
+     * Patches a directRequestProcess.
+     *
+     * @param entryId entry of the directRequestProcess
+     * @param dto updater containing the amount of openSlots (int)
+     *
+     * @return updated directRequestProcess
+     */
+    @PutMapping(value = "/categories/{category_id}/entries/{entry_id}/process", produces = "application/json")
```
Comment (Wil-Lee) [inline src/main/java/com/peerrequest/app/api/DirectRequestsController.java:80]: remove:  produce = "aplication/json"
to match the other controllers

File: src/main/java/com/peerrequest/app/api/DirectRequestsController.java [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+    }
+
+    /**
+     * Gets a directRequests.
+     *
+     * @param requestId request to get
+     *
+     * @return directRequest with requestId
+     */
+    @GetMapping(value = "/categories/{category_id}/entries/{entry_id}/process/requests/{request_id}",
```
Comment (Wil-Lee) [inline src/main/java/com/peerrequest/app/api/DirectRequestsController.java:117]: remove:  produce = "aplication/json"
to match the other controllers

File: src/main/java/com/peerrequest/app/api/DirectRequestsController.java [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+        return option.get().toDto();
+    }
+
+    /**
+     * Gets the list of all requests of an entry.
+     *
+     * @return directRequest with requestId
+     */
+    @GetMapping(value = "/categories/{category_id}/entries/{entry_id}/process/requests",
+            produces = "application/json")
```
Comment (Wil-Lee) [inline src/main/java/com/peerrequest/app/api/DirectRequestsController.java:147]: remove:  produce = "aplication/json"
to match the other controllers

**Review by Wil-Lee — CHANGES_REQUESTED** [review]

File: src/main/java/com/peerrequest/app/api/DirectRequestsController.java [hunk 9]
```diff
@@ -0,0 +1,264 @@
...
+
+    /**
+     * Patches a directRequest.
+     *
+     * @param requestId DirectRequest to patch
+     * @param updater updater containing a DirectRequest.State
+     *
+     * @return the updated request
+     */
+    @PatchMapping(value = "/categories/{category_id}/entries/{entry_id}/process/requests/{request_id}",
```
Comment (Wil-Lee) [inline src/main/java/com/peerrequest/app/api/DirectRequestsController.java:189]: Wrong endpoint. remove /{request_id} and get the id of the request from the requestBody.
It works like this too but we did it already different in the other controllers

File: src/main/java/com/peerrequest/app/api/DirectRequestsController.java [hunk 10]
```diff
@@ -0,0 +1,264 @@
...
+            produces = "application/json")
```
Comment (Wil-Lee) [inline src/main/java/com/peerrequest/app/api/DirectRequestsController.java:190]: remove produces = "application/json"

**Review by Wil-Lee — DISMISSED** [review]

File: src/main/java/com/peerrequest/app/api/DirectRequestsController.java (hunk 3 above)
Comment (okidokidokkaebi) [reply src/main/java/com/peerrequest/app/api/DirectRequestsController.java:37]: Not necessary, this information should be public to authenticated users.

File: src/main/java/com/peerrequest/app/api/DirectRequestsController.java (hunk 4 above)
Comment (okidokidokkaebi) [reply src/main/java/com/peerrequest/app/api/DirectRequestsController.java:31]: Done.

File: src/main/java/com/peerrequest/app/api/DirectRequestsController.java (hunk 5 above)
Comment (okidokidokkaebi) [reply src/main/java/com/peerrequest/app/api/DirectRequestsController.java:49]: Done.

File: src/main/java/com/peerrequest/app/api/DirectRequestsController.java (hunk 6 above)
Comment (okidokidokkaebi) [reply src/main/java/com/peerrequest/app/api/DirectRequestsController.java:80]: Done.

File: src/main/java/com/peerrequest/app/api/DirectRequestsController.java (hunk 7 above)
Comment (okidokidokkaebi) [reply src/main/java/com/peerrequest/app/api/DirectRequestsController.java:117]: Done.

File: src/main/java/com/peerrequest/app/api/DirectRequestsController.java (hunk 8 above)
Comment (okidokidokkaebi) [reply src/main/java/com/peerrequest/app/api/DirectRequestsController.java:147]: Done.

File: src/main/java/com/peerrequest/app/api/DirectRequestsController.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (okidokidokkaebi) [reply src/main/java/com/peerrequest/app/api/DirectRequestsController.java:80]: Done.

File: src/main/java/com/peerrequest/app/api/DirectRequestsController.java (hunk 9 above)
Comment (okidokidokkaebi) [reply src/main/java/com/peerrequest/app/api/DirectRequestsController.java:189]: Done.

File: src/main/java/com/peerrequest/app/api/DirectRequestsController.java (hunk 10 above)
Comment (okidokidokkaebi) [reply src/main/java/com/peerrequest/app/api/DirectRequestsController.java:190]: Done.

**Review by Wil-Lee — APPROVED** [review]

**Review by Wil-Lee — APPROVED** [review]
Is not finished yet but he wanted me to give my consent :(((((((

**Pull request merged**